File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 3/1/2026, 11:44:48 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        maxPrice = 0
        minPrice = float('inf')
        
        for price in prices:
            if(price < minPrice):
                minPrice = price
            
            else:
                currentPrice = price - minPrice

                if(currentPrice>maxPrice):
                    maxPrice = currentPrice

        return maxPrice

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.js
// Last updated: 3/1/2026, 11:44:45 PM
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minPrice = Infinity
    let maxProfit = 0
    for (let num of prices){
        if(num < minPrice){
            minPrice = num
        }
         let profit = num - minPrice
        if(profit > maxProfit){
            maxProfit = profit
        }
    }return maxProfit
    
};